dist/c/include/qiskit/complex.h

```diff
@@ -0,0 +1,84 @@
+#ifndef QISKIT__COMPLEX_H
+#define QISKIT__COMPLEX_H
+/**
+ * A complex double.
+ *
+ * See also ``qk_complex64_to_native`` and ``qk_complex64_from_native`` to convert
+ * this struct to (or from) a compiler-native complex number representation.
+ */
+typedef struct {
+    /* The real part. */
+    double re;
+    /* The imaginary part. */
+    double im;
+} QkComplex64;
+
+// Complex number typedefs conversions.
+#ifdef __cplusplus
+#include <complex>
+static std::complex<double> qk_complex64_to_native(QkComplex64 *value) {
```

Comment:
Should we include docstrings for each conversion function? Or will doxygen generate too many definitions?

Reply:
I updated the docs to implement the above 🙂